# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/Notification.cpp
# Commit: ed71fdc2d78d
# Full Hash: ed71fdc2d78dec085a3d153220bba4aeda89992b
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-03-05 04:28:59
# Regressor Bug: 1951379
# File Overlap Count: 1
# Description:
#   Bug 1951379 - Part 3: Add NotificationStorageEntry r=asuth
#   
#   This skips handling `vibrate` for now, as we don't ship it and it also complicates test setup.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240213
# ==============================================================================

diff -r 61ab7888abc4 -r ed71fdc2d78d dom/notification/Notification.cpp
--- a/dom/notification/Notification.cpp	Tue Mar 04 18:00:17 2025 +0000
+++ b/dom/notification/Notification.cpp	Tue Mar 04 18:00:18 2025 +0000
@@ -377,6 +377,9 @@
   options.mBody = ipcOptions.body();
   options.mTag = ipcOptions.tag();
   options.mIcon = ipcOptions.icon();
+  options.mRequireInteraction = ipcOptions.requireInteraction();
+  options.mSilent = ipcOptions.silent();
+
   IgnoredErrorResult rv;
   RefPtr<Notification> notification =
       CreateInternal(aGlobal, aIPCNotification.id(), ipcOptions.title(),